### PR TITLE
Added namespaces to all namespace-scoped objects

### PR DESCRIPTION
**Pull request by dlydiard**
#### What this PR does / why we need it:
Added namespaces to all namespace scoped objects using the HELM standard Release.namespace.

#### Which issue this PR fixes
When installing this chart via Kustomize and syncing with ArgoCD, the chart does not get installed into the intended target namespace. This change should not affect existing behavior while still allowing for additional tooling to operate with the chart.

#### Special notes for your reviewer:
This is a resubmittal of https://github.com/DataDog/helm-charts/pull/723 since the original code owner isn't available to make the requested additional changes. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated

### CODE REVIEW

**Review by clamoriniere — APPROVED** [review]
Hi @dlydiard 
thanks for submitting this pull-requests.

The changes looks good to me. I will run the CI and merge it. 
Cedric

File: charts/datadog-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "datadog-operator.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (clamoriniere) [inline charts/datadog-operator/templates/deployment.yaml:5]: the file is part of the `datadog-operator` chart not the `datadog` chart. 
the change still make sense, but it require to also:
* bump the `datadog-operator` chart version
* update the `datadog-operator` changelog.md file.

**Review by clamoriniere — CHANGES_REQUESTED** [review]
Sorry I was to fast during my initial review.

the template files that you are modifying are in the `datadog-operator` chart. but you have modified the `datadog` chart metadata file `chart.yaml` and `changelog.md`

could you please:
* revert the change made in the `datadog` chart
* update  `chart.yaml` and `changelog.md` in `datadog-operator` chart.

File: charts/datadog-operator/templates/deployment.yaml (hunk 1 above)
Comment (dlydiard) [reply charts/datadog-operator/templates/deployment.yaml:5]: updated, ty!

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**